tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,382 @@
+#   not nice.
+# - Alternatively, one can subset trials already at the epoch level using MNEs
+#   event information, and create individual PSD and SNR objects.
+#
+# Here we have already subsetted trials before snr calculation (only 12Hz
+# stimulation) and will now compare SNR in different channel subsets.
+#
+# For illustration purposes, we will still subset the first 5 and last 5 of the
+# 10 trials with 12hz stimulation.
+
+
+##############################################################################
+# Define ROIs
+roi_temporal = ['T7', 'F7', 'T8', 'F8']  # temporal
+roi_aud = ['AFz', 'Fz', 'FCz', 'Cz', 'CPz', 'F1', 'FC1',
+           'C1', 'CP1', 'F2', 'FC2', 'C2', 'CP2']  # auditory roi
+roi_vis = ['POz', 'Oz', 'O1', 'O2', 'PO3', 'PO4', 'PO7',
+           'PO8', 'PO9', 'PO10', 'O9', 'O10']  # visual roi
+
+##############################################################################
+# Create corresponding picks
+picks_roi_temp = mne.pick_types(raw.info, eeg=True, stim=False,
+                                exclude='bads', selection=roi_temporal)
+picks_roi_aud = mne.pick_types(raw.info, eeg=True, stim=False,
+                               exclude='bads', selection=roi_aud)
+picks_roi_vis = mne.pick_types(raw.info, eeg=True, stim=False,
+                               exclude='bads', selection=roi_vis)
+
+##############################################################################
+# Subset data based on ROIs
+snrs_trialwise_roi_aud = snrs_stim[:, picks_roi_aud]
+snrs_trialwise_roi_vis = snrs_stim[:, picks_roi_vis]
+snrs_trialwise_temp = snrs_stim[:, picks_roi_temp]
+
+##############################################################################
+# SNR for different ROIs
+print('mean SNR (all channels, all trials) at %iHz = %.3f '
+      % (stim_freq, snrs_stim.mean()))
+print('mean SNR (auditory ROI) at %iHz = %.3f '
+      % (stim_freq, snrs_trialwise_roi_aud.mean()))
+print('mean SNR (visual ROI) at %iHz = %.3f '
+      % (stim_freq, snrs_trialwise_roi_vis.mean()))
+print('mean SNR (temporal chans) at %iHz = %.3f '
+      % (stim_freq, snrs_trialwise_temp.mean()))
+
+
+##############################################################################
+# Define trial subsets
+i_cat1_1 = [i for i in range(5)]
+i_cat1_2 = [i for i in range(5, 10)]
+
+##############################################################################
+# Subset data trial-wise
+snrs_trialwise_cat1_1 = snrs_stim[i_cat1_1, :]
+snrs_trialwise_cat1_2 = snrs_stim[i_cat1_2, :]
+
+##############################################################################
+# SNR for different subsets of trials
+print('mean SNR (trial subset 1) at %iHz = %.3f '
+      % (stim_freq, snrs_trialwise_cat1_1.mean()))
+print('mean SNR (trial subset 2) at %iHz = %.3f '
+      % (stim_freq, snrs_trialwise_cat1_2.mean()))
+
+
+##############################################################################
+# Statistics
+# ----------
+# Just a toy t-test example to test whether:
```

Comment:
i'd rather like to show an analysis that actually makes sense :)
dont think, roi comparison really does in sensor space EEG.. 

i dont have a better idea though, ad hoc... do you?!

Reply:
I agree, it does not make sense. I say we stick to comparing SNR in different conditions.